app/models/credit_card.rb

```diff
@@ -1,14 +1,40 @@
 class CreditCard < ApplicationRecord
-  before_save :set_last_four
+  include ActiveModel::Validations
+
+  attr_accessor :card_number
+  attr_accessor :cvv
 
-  attr_accessor :number
+  before_save :set_last_four
+  validates :card_number, presence: true, credit_card_number: true
+  validates :cvv, :first_name, :last_name, :month, :year, presence: true
+  validates :cvv, length: { is: 3 }
+  validate :expiry_cannot_be_in_the_past
 
-  validates :first_name, :last_name, :month, :year, presence: true
   belongs_to :user
 
+  def expired?
+    expired_by?(Time.zone.now.to_date)
+  end
+
+  def expired_by?(date)
+    expiration_date < date
+  end
+
+  def expiration_date
+    DateTime.parse("#{self.year}-#{self.month}-01").end_of_month
```

Comment:
@ksolkowski i got this from wantable-2. What's the difference between the two? From what I read they're basically the same, but ActiveRecord prefers DateTime (not that it matters here).